Packs/EWS/Integrations/EWSv2/EWSv2.py

```diff
@@ -1241,33 +1239,51 @@ def parse_incident_from_item(item, is_fetch):
 
 def fetch_emails_as_incidents(account_email, folder_name):
     last_run = get_last_run()
+    excluded_ids = set(last_run.get(LAST_RUN_IDS, []))
 
     try:
         account = get_account(account_email)
         last_emails = fetch_last_emails(account, folder_name, last_run.get(LAST_RUN_TIME), last_run.get(LAST_RUN_IDS))
 
-        ids = deque(last_run.get(LAST_RUN_IDS, []), maxlen=LAST_RUN_IDS_QUEUE_SIZE)
         incidents = []
         incident = {}  # type: Dict[Any, Any]
+        current_fetch_ids = set()
+
         for item in last_emails:
             if item.message_id:
-                ids.append(item.message_id)
+                current_fetch_ids.add(item.message_id)
                 incident = parse_incident_from_item(item, True)
                 if incident:
                     incidents.append(incident)
 
                 if len(incidents) >= MAX_FETCH:
                     break
 
-        last_run_time = incident.get('occurred', last_run.get(LAST_RUN_TIME))
-        if isinstance(last_run_time, EWSDateTime):
-            last_run_time = last_run_time.ewsformat()
+        demisto.debug('EWS V2 - ending fetch - got {} incidents.'.format(len(incidents)))
+        last_fetch_time = last_run.get(LAST_RUN_TIME)
+
+        last_incident_run_time = incident.get("occurred", last_fetch_time)
```

Comment:
same question about its, ordering as in O365

Reply:
yes
![image](https://user-images.githubusercontent.com/71636766/126793004-eb423abc-adea-4999-9522-7dfa76bb014b.png)